watertap/core/membrane_channel0d.py

```diff
@@ -0,0 +1,262 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+from pyomo.environ import (
+    NonNegativeReals,
+    NegativeReals,
+    Param,
+    Set,
+    Var,
+    value,
+    units as pyunits,
+)
+from idaes.core import (
+    declare_process_block_class,
+    EnergyBalanceType,
+    FlowDirection,
+)
+from idaes.core.util import scaling as iscale
+from idaes.core.util.misc import add_object_reference
+from idaes.core.util.exceptions import BalanceTypeNotSupportedError
+from idaes.core.base.control_volume0d import ControlVolume0DBlockData
+import idaes.logger as idaeslog
+
+from watertap.core.membrane_channel_base import (
+    MembraneChannelMixin,
+    PressureChangeType,
+    CONFIG_Template,
+)
+
+
+@declare_process_block_class("MembraneChannel0DBlock")
+class MembraneChannel0DBlockData(MembraneChannelMixin, ControlVolume0DBlockData):
+
+    # overwrite CV0D `add_geometry`
+    def add_geometry(self, include_length_and_width=True):
+        """
+        Method to create spatial domain and volume Var in ControlVolume.
+
+        Args:
+            include_length_and_width - (optional) add a length and width
+                variables to the membrane channel. Default: `True`
+
+        Returns:
+            None
+        """
+
+        if include_length_and_width:
+            units_meta = self.config.property_package.get_metadata().get_derived_units
+            self.length = Var(
+                initialize=10,
+                bounds=(0.1, 5e2),
+                domain=NonNegativeReals,
+                units=units_meta("length"),
+                doc="Effective membrane length",
+            )
+            self.width = Var(
+                initialize=1,
+                bounds=(1e-1, 1e3),
+                domain=NonNegativeReals,
+                units=units_meta("length"),
+                doc="Membrane width",
+            )
+
+    def add_state_blocks(
+        self, information_flow=FlowDirection.forward, has_phase_equilibrium=None
+    ):
+        """
+        This method constructs the state blocks for the
+        control volume.
+
+        Args:
+            information_flow: a FlowDirection Enum indicating whether
+                               information flows from inlet-to-outlet or
+                               outlet-to-inlet
```

Comment:
This look like it is referring to a feature in the core IDAES code, which allows for systems where it is more robust to solve a process in reverse order (e.g. some refrigeration cycles). It is not related (directly) to co- or counter-current flow.

I am not sure whether this feature is required here, and unless there is a need for it I would suggest it might be best to drop it.

Reply:
Right, the only reason it is here is for the call to `super().add_state_blocks` -- I gave the overloaded methods the same signature as that in the control volume. However, if we think for membrane channels we'll always use the default, I'm happy to eliminate the option.

Reply:
Oh its not related to co- or counter-current flow? I think it can be eliminated then, although I don't know anything about the benefits of solving a process in reverse order (e.g. Andrew's comment about refrigeration cycles).

Reply:
This feature has never actually been used in IDAES - it was added based on a suggestion/request from a team member for future applications.

Reply:
I've remove it as part of 495fbfa9